### PR TITLE
Allow Panamax to execute jobs via Dray

### PR DESCRIPTION
**Pull request by dharmamike**
This pull request introduces several new items into Panamax, all around the concept of running jobs via CenturyLinkLabs/Dray.  

Workflows to be executed are stored in YAML template files read in as `JobTemplate` subclass instances with the `panamax:job_templates:load` rake task.  Currently the only subclass is `ClusterJobTemplate`. The templates are parsed so that each step of the workflow is added as a `JobTemplateStep` for that `JobTemplate`.

By POSTing to the new `/jobs` route, a `Job` can be created from a `JobTemplate` and subsequently posted to a running Dray service.  The Dray service will return a job key, and subsequent GET requests to the `/jobs/:key` route will return details of the job by fetching the job data from Dray.

### CODE REVIEW

File: app/builders/job_builder/from_template.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+module JobBuilder
+  class FromTemplate
+
+    def self.create_job(options)
+      fail 'no template id given' unless options[:template_id]
+
+      template = JobTemplate.find(options[:template_id])
+      if options[:override]
+        override = JobTemplateBuilder.create(options[:override], false)
+        template.override(override) if override
```
Comment (dharmamike) [inline app/builders/job_builder/from_template.rb]: No need to check `if override`

File: lib/panamax_agent/configuration.rb [hunk 2]
```diff
@@ -28,6 +29,7 @@ module Configuration
     DEFAULT_JOURNAL_API_URL = ENV['JOURNAL_ENDPOINT']
     DEFAULT_PMX_REGISTRY_API_URL = 'http://74.201.240.198:5000'
     DEFAULT_PMX_REGISTRY_API_VERSION = 'v1'
+    DEFAULT_STEVEDORE_API_URL = ENV['STEVEDORE_ENDPOINT']
```
Comment (bdehamer) [inline lib/panamax_agent/configuration.rb]: Stevedore -> Dray

**Comment by bdehamer** [comment]
Found another occurrence of "stevedore", but fixing that and merging Alex's fixes from https://github.com/CenturyLinkLabs/panamax-api/pull/262 == :+1:

**Comment by alexwelch** [comment]
:+1:

**Comment by alexwelch** [comment]
@dharmamike feel free to slice and dice my commits as you see fit before merging

**Pull request merged**